### PR TITLE
feat(1082): [2] add get branch list function

### PR DESCRIPTION
**Pull request by yuichi10**
## Context
To get pipelines, we need a function which get list of branch.  

## Objective
Add getBranchList function which will return branch list.

## References
Blocked by https://github.com/screwdriver-cd/data-schema/pull/240
Related to screwdriver-cd/screwdriver#1082

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -97,6 +97,23 @@ class ScmBase {
         return Promise.reject(new Error('Not implemented'));
     }
 
+    /**
+     * Get the branch list which related the repository
```
Comment (tkyi) [inline index.js]: Get the branch list ~which~ related `to` the repository

File: README.md [hunk 2]
```diff
@@ -379,6 +379,21 @@ The received webhook is available or not as boolean.
...
+| :-------------   | :---- | :------- | :-------------|
+| config        | Object | Yes | Configuration Object |
+| config.scmUri | String | Yes | The scm uri (ex: `github.com:1234:branchName`) |
+| config.token | String | Yes | Access token for scm |
+
+#### Expected Outcome
+The array of object consists of branch name.
+
+#### Expected Promise Response
+1. Resolve with the array of object consists of branch name.
```
Comment (tkyi) [inline README.md]: Resolve with ~the~ `an` array of object`s` consist~s~`ing` of branch names.

File: index.js (hunk 1 above)
Comment (yuichi10) [reply index.js]: Thank you! I will fix.

File: README.md (hunk 2 above)
Comment (yuichi10) [reply README.md]: I will fix.

**Comment by yuichi10** [comment]
@tkyi Would somebody check this PR? Thank you.

**Pull request merged**